daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala

```diff
@@ -911,6 +910,64 @@ class DecodeV1Spec
       }
     }
 
+    val interfacePrimitivesDottedNameTable = ImmArraySeq("Mod", "I").map(Ref.DottedName.assertFromString)
+
+    val interfacePrimitivesTestCases = {
+
+      val unit = DamlLf1.Unit.newBuilder().build()
+      val pkgRef = DamlLf1.PackageRef.newBuilder().setSelf(unit).build
+      val modRef = DamlLf1.ModuleRef.newBuilder().setPackageRef(pkgRef).setModuleNameInternedDname(0).build()
+      val ifaceTyConName = DamlLf1.TypeConName.newBuilder().setModule(modRef).setNameInternedDname(1)
+      val scalaIfaceTyConName = Ref.TypeConName.assertFromString("noPkgId:Mod:I")
+
+      val interfaceTemplateTypeRep = DamlLf1.Expr.newBuilder().setInterfaceTemplateTypeRep(
+        DamlLf1.Expr.InterfaceTemplateTypeRep.newBuilder()
+          .setInterface(ifaceTyConName)
+          .setExpr(unitExpr)
+          .build()
+      ).build()
+
+      val signatoryInterface = DamlLf1.Expr.newBuilder().setSignatoryInterface(
+        DamlLf1.Expr.SignatoryInterface.newBuilder()
+          .setInterface(ifaceTyConName)
+          .setExpr(unitExpr)
+          .build()
+      ).build()
+
+      val observerInterface = DamlLf1.Expr.newBuilder().setObserverInterface(
+        DamlLf1.Expr.ObserverInterface.newBuilder()
+          .setInterface(ifaceTyConName)
+          .setExpr(unitExpr)
+          .build()
+      ).build()
+
+      Table(
+        "input" -> "expected output",
+        interfaceTemplateTypeRep -> EInterfaceTemplateTypeRep(ifaceId = scalaIfaceTyConName, body = EUnit),
+        signatoryInterface -> ESignatoryInterface(ifaceId = scalaIfaceTyConName, body = EUnit),
+        observerInterface -> EObserverInterface(ifaceId = scalaIfaceTyConName, body = EUnit),
+      )
+    }
+
+    val interfacePrimitivesDecoder = (version: LV) =>
+      moduleDecoder(version, dottedNameTable = interfacePrimitivesDottedNameTable)
+
+    s"reject interface_template_type_rep if version < ${LV.Features.interfaces}" in {
+      forEveryVersionSuchThat(_ <= LV.Features.interfaces) { version =>
+        forEvery(interfacePrimitivesTestCases) { (proto, _) =>
+          an[Error.Parsing] shouldBe thrownBy(interfacePrimitivesDecoder(version).decodeExpr(proto, "test"))
```

Comment:
for some reason when I run this locally I get

```
tableDrivenForEveryFailed:   org.scalatest.exceptions.TableDrivenPropertyCheckFailedException: TestFailedException was thrown during property evaluation. (DecodeV1Spec.scala:957)
      Message: Expected exception com.daml.lf.archive.Error$Parsing to be thrown, but no exception was thrown
      Location: (DecodeV1Spec.scala:958)
```

but if I change the body of the loop to match the `accept` test below I get

```
  tableDrivenForEveryFailed:   org.scalatest.exceptions.TableDrivenPropertyCheckFailedException: Parsing was thrown during property evaluation. (DecodeV1Spec.scala:957)
      Message: Expr.interface_template_type_rep is not supported by Daml-LF 1.LanguageMinorVersion(6)
```

so which one is it?